packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit_api_impls.dart

```diff
@@ -10,52 +10,50 @@ import '../common/web_kit.pigeon.dart';
-  return types.map<WKWebsiteDataTypesEnumData>((WKWebsiteDataTypes type) {
-    late final WKWebsiteDataTypesEnum value;
+Iterable<WKWebsiteDataTypeEnumData> _toWKWebsiteDataTypeEnumData(
+    Iterable<WKWebsiteDataType> types) {
+  return types.map<WKWebsiteDataTypeEnumData>((WKWebsiteDataType type) {
+    late final WKWebsiteDataTypeEnum value;
     switch (type) {
-      case WKWebsiteDataTypes.cookies:
-        value = WKWebsiteDataTypesEnum.cookies;
+      case WKWebsiteDataType.cookies:
+        value = WKWebsiteDataTypeEnum.cookies;
         break;
-      case WKWebsiteDataTypes.memoryCache:
-        value = WKWebsiteDataTypesEnum.memoryCache;
+      case WKWebsiteDataType.memoryCache:
+        value = WKWebsiteDataTypeEnum.memoryCache;
         break;
-      case WKWebsiteDataTypes.diskCache:
-        value = WKWebsiteDataTypesEnum.diskCache;
+      case WKWebsiteDataType.diskCache:
+        value = WKWebsiteDataTypeEnum.diskCache;
         break;
-      case WKWebsiteDataTypes.offlineWebApplicationCache:
-        value = WKWebsiteDataTypesEnum.offlineWebApplicationCache;
+      case WKWebsiteDataType.offlineWebApplicationCache:
+        value = WKWebsiteDataTypeEnum.offlineWebApplicationCache;
         break;
-      case WKWebsiteDataTypes.localStroage:
-        value = WKWebsiteDataTypesEnum.localStroage;
+      case WKWebsiteDataType.localStorage:
+        value = WKWebsiteDataTypeEnum.localStorage;
         break;
-      case WKWebsiteDataTypes.sessionStorage:
-        value = WKWebsiteDataTypesEnum.sessionStorage;
+      case WKWebsiteDataType.sessionStorage:
+        value = WKWebsiteDataTypeEnum.sessionStorage;
         break;
-      case WKWebsiteDataTypes.sqlDatabases:
-        value = WKWebsiteDataTypesEnum.sqlDatabases;
+      case WKWebsiteDataType.webSQLDatabases:
+        value = WKWebsiteDataTypeEnum.webSQLDatabases;
         break;
-      case WKWebsiteDataTypes.indexedDBDatabases:
-        value = WKWebsiteDataTypesEnum.indexedDBDatabases;
+      case WKWebsiteDataType.indexedDBDatabases:
+        value = WKWebsiteDataTypeEnum.indexedDBDatabases;
         break;
     }
 
-    return WKWebsiteDataTypesEnumData(value: value);
+    return WKWebsiteDataTypeEnumData(value: value);
   });
 }
 
 extension _NSHttpCookieConverter on NSHttpCookie {
   NSHttpCookieData toNSHttpCookieData() {
     return NSHttpCookieData(
-      properties: properties.map<NSHttpCookiePropertyKeyEnumData, String>(
-        (NSHttpCookiePropertyKey key, Object value) {
-          return MapEntry<NSHttpCookiePropertyKeyEnumData, String>(
-            key.toNSHttpCookiePropertyKeyEnumData(),
-            value.toString(),
-          );
+      propertyKeys: properties.keys.map<NSHttpCookiePropertyKeyEnumData>(
+        (NSHttpCookiePropertyKey key) {
+          return key.toNSHttpCookiePropertyKeyEnumData();
         },
-      ),
+      ).toList(),
+      propertyValues: properties.values.map<Object>((_) => _).toList(),
```

Comment:
It's not clear to me from the docs that Dart map orders for unordered implementations have to be consistent, so this makes me nervous.

I would rather you extract `properties.keys` to a local `Iterable`, and then use it as the driver for both map operations. It'll be a little less efficient, but guaranteed to be correct.

Reply:
Oops, I was not looking at the right part of the docs. That does indeed address my concern.

> I'm not sure what you mean by unordered implementations

I was talking about this part of the docs:
> The plain [HashMap](https://api.dart.dev/stable/2.17.0/dart-collection/HashMap-class.html) is unordered (no order is guaranteed),
the [LinkedHashMap](https://api.dart.dev/stable/2.17.0/dart-collection/LinkedHashMap-class.html) iterates in key insertion order,
and a sorted map like [SplayTreeMap](https://api.dart.dev/stable/2.17.0/dart-collection/SplayTreeMap-class.html) iterates the keys in sorted order.

My concern was that for `HashMap` there might not be any guarantees *at all* about order, including between calling `keys` and then calling `values` afterward. But that would violate the `values` docs you referenced.